specs/experimental/op-stack-manager.md

```diff
@@ -0,0 +1,214 @@
+
+The OP Stack Manager is a proxied contract deployed at `0xTODO`. It can be deployed as follows:
+
+TODO.
+
+## Interface
+
+Version 1.0.0 of the OP Stack Manager deploys the [`op-contracts/v1.4.0`]
+contracts release.
+
+### `Proxy.sol`
+
+The OP Stack Manager is a proxied contract using the standard `Proxy.sol` contract that lives in
+the Optimism monorepo. Therefore the OP Stack Manager will have the same interface as the
+`Proxy.sol`, in addition to other methods defined in this specification.
+
+The privileged methods of the OP Stack Manager will be held by the L1 ProxyAdmin owner, as
+specified by the [standard configuration].
+
+### `deploy`
+
+The `deploy` method is the only non-view method in the contract. It is used to
+deploy the full set of L1 contracts required to setup a new OP Stack chain that
+complies with the [standard configuration]. It has the following interface:
+
+```solidity
+struct Roles {
+  address proxyAdminOwner;
+  address systemConfigOwner;
+  bytes32 batcherHash;
+  address unsafeBlockSigner;
+  address proposer;
+  address challenger;
+}
+
+function deploy(
+  uint64 l2ChainId,
+  Roles roles,
+  uint256 scalar
+) external returns (SystemConfig)
+```
+
+The `l2ChainId` has the following restrictions:
+
+- It must not be equal to 0.
+- It must not be equal to the chain ID of the chain the OP Stack Manager is
+deployed on.
+- It must not be equal to a chain ID that is already present in the
+[ethereum-lists/chains] repository. This is not enforced onchain, but may matter
+for future versions of OP Stack Manager that handle upgrades.
+
+On success, the following event is emitted:
+
+```solidity
+event Deployed(uint256 indexed l2ChainId, SystemConfig indexed systemConfig);
+```
+
+This method reverts on failure. This occurs when:
+
+- The input `l2ChainId` does not comply with the restrictions above.
+- The resulting configuration is not compliant with the [standard configuration].
+
+### Getter Methods
+
+The following interface defines the available getter methods:
+
+```solidity
+/// @notice The logic address and initializer selector for an implementation contract.
+struct Implementation {
+  address logic; // Address containing the deployed logic contract.
+  bytes4 initializer; // Function selector for the initializer.
+}
+
+/// @notice Returns the latest approved release of the OP Stack contracts.
+/// @notice Release strings follow semver and are named with the
+/// format `op-contracts/vX.Y.Z`.
+function latestRelease() external view returns (string memory);
+
+/// @notice Maps a release version to a contract name to its implementation data.
+function implementation(
+  string memory release,
+  string memory contractName
+) external view returns (Implementation memory);
+
+/// @notice Maps an L2 Chain ID to the SystemConfig address for that chain.
+/// @notice Most information for a chain can be found from its SystemConfig,
+/// with ProtocolVersions being the exception. ProtocolVersions is a singleton
```

Comment:
I think the mention of `ProtocolVersions` is unnecessary details here, we should add a getter to the `SuperchainConfig` for `ProtocolVersions`

Reply:
Agreed, my understanding is that it's planned as part of https://github.com/ethereum-optimism/client-pod/issues/699#issuecomment-2150970346. Tweaked this to remove the PV reference in 8e57051